plugins/hls-rename-plugin/src/Ide/Plugin/Rename.hs

```diff
@@ -70,8 +71,8 @@ renameProvider state pluginId (RenameParams _prog docId@(TextDocumentIdentifier
            indirect references through punned names. To find the transitive closure, we do a pass of
            the direct references to find the references for any punned names.
            See the `IndirectPuns` test for an example. -}
-        indirectOldNames <- concat . filter ((>1) . Prelude.length) <$>
-            mapM (uncurry (getNamesAtPos state) . locToFilePos) directRefs
+        indirectOldNames <- concat . filter notNull <$>
+            mapM (uncurry (getNamesAtPos state)) (mapMaybe locToFilePos directRefs)
```

Comment:
Change locToFilePos (see def. below) to return maybe.
I hope that silently ignoring paths for which uriToNormalizedFilePath returns Nothing, rather than crashing is ok.

Reply:
Hmm, I think we might want to fail the rename, because that means there's a potential reference that we couldn't resolve, so we might not be doing a correct rename.

Reply:
Maybe just add a comment at least.

Reply:
Ok, will dive a bit into how this works and get back to this.

Reply:
Ok, I found there's a dedicated `getNormalizedFilePathE` function for this that just translates the Nothing to PluginError, which sounds cleaner/more informative than pattern match error..